### PR TITLE
[Documentation] bumped maintenance branches in docs

### PR DESCRIPTION
**Pull request by foxou33**
| Q               | A
| --------------- | -----
| Branch?         | 1.7
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT

### CODE REVIEW

File: docs/contributing/code/patches.rst [hunk 1]
```diff
@@ -100,12 +100,12 @@ Choose the right Base Branch
 
 Before starting to work on a patch, you must determine on which branch you need to work. It will be:
 
-* ``1.4`` or ``1.5``, if you are fixing a bug for an existing feature or want to make a change that falls into the list of acceptable changes in patch versions
+* ``1.6`` or ``1.7``, if you are fixing a bug for an existing feature or want to make a change that falls into the list of acceptable changes in patch versions
```
Comment (lchrusciel) [inline docs/contributing/code/patches.rst:103]:
```suggestion
* ``1.7``, if you are fixing a bug for an existing feature or want to make a change that falls into the list of acceptable changes in patch versions
```

File: docs/contributing/code/patches.rst [hunk 2]
```diff
@@ -100,12 +100,12 @@ Choose the right Base Branch
...
 * ``master``, if you are adding a new feature.
 
 .. note::
 
-    All bug fixes merged into the ``1.4`` and ``1.5`` maintenance branches are also merged into ``master`` on a regular basis.
+    All bug fixes merged into the ``1.6`` and ``1.7`` maintenance branches are also merged into ``master`` on a regular basis.
```
Comment (lchrusciel) [inline docs/contributing/code/patches.rst:108]:
```suggestion
    All bug fixes merged into the ``1.7`` maintenance branches are also merged into ``master`` on a regular basis.
```

File: docs/contributing/code/patches.rst [hunk 3]
```diff
@@ -246,7 +246,7 @@ possible:
 
     | Q               | A
     | --------------- | -----
-    | Branch?         | 1.4, 1.5 or master
+    | Branch?         | 1.6, 1.7 or master
```
Comment (lchrusciel) [inline docs/contributing/code/patches.rst:249]:
```suggestion
    | Branch?         | 1.7 or master
```

File: docs/contributing/code/patches.rst [hunk 4]
```diff
@@ -260,7 +260,7 @@ An example submission could now look as follows:
 
     | Q               | A
     | --------------- | -----
-    | Branch?         | 1.4
+    | Branch?         | 1.6
```
Comment (lchrusciel) [inline docs/contributing/code/patches.rst:263]:
```suggestion
    | Branch?         | 1.7
```

File: docs/contributing/documentation/overview.rst [hunk 5]
```diff
@@ -38,12 +38,12 @@ Choose the right Base Branch
 
 Before starting to work on a patch, you must determine on which branch you need to work. It will be:
 
-* ``1.4`` or ``1.5``, if you are fixing or adding docs for features that exist in one of those versions,
-* ``master``, if you are documenting a new feature, that was not in ``1.4`` nor in ``1.5``
+* ``1.6`` or ``1.7``, if you are fixing or adding docs for features that exist in one of those versions,
```
Comment (lchrusciel) [inline docs/contributing/documentation/overview.rst:41]:
```suggestion
* ``1.7``, if you are fixing or adding docs for features that exist in one of those versions,
```

File: docs/contributing/documentation/overview.rst [hunk 6]
```diff
@@ -38,12 +38,12 @@ Choose the right Base Branch
...
+* ``master``, if you are documenting a new feature, that was not in ``1.6`` nor in ``1.7``
```
Comment (lchrusciel) [inline docs/contributing/documentation/overview.rst:42]:
```suggestion
* ``master``, if you are documenting a new feature, that was not in ``1.7``
```

File: docs/contributing/documentation/overview.rst [hunk 7]
```diff
@@ -38,12 +38,12 @@ Choose the right Base Branch
...
 
 .. note::
 
-    All bug fixes merged into the ``1.4`` and ``1.5`` maintenance branches are also merged into ``master`` on a regular basis.
+    All bug fixes merged into the ``1.6`` and ``1.7`` maintenance branches are also merged into ``master`` on a regular basis.
```
Comment (lchrusciel) [inline docs/contributing/documentation/overview.rst:46]:
```suggestion
    All bug fixes merged into the ``1.7`` maintenance branches are also merged into ``master`` on a regular basis.
```

**Comment by foxou33** [comment]
I tried to squash the last 4 commits but I think I messed up ?

**Comment by lchrusciel** [comment]
Hey @foxou33,

another rebase with 1.7 branch was needed. I've done it for you. Also updated a few other branches' names in the meantime. I hope you don't mind. Once the PR will be green it is good to go :)

**Comment by lchrusciel** [comment]
Thanks, @foxou33! :tada:

**Pull request merged**